apps/dashboard/src/components/GenericSearch.tsx

```diff
@@ -0,0 +1,26 @@
+import React from "react"
+import { Autocomplete } from "@mantine/core"
+
+interface GenericSearchProps<T> {
+  onSearch: (query: string) => void
+  onSubmit: (item: T) => void
+  items: T[]
+  dataMapper: (item: T) => string
+}
+
+const GenericSearch = <T,>({ onSearch, onSubmit, items, dataMapper }: GenericSearchProps<T>) => {
```

Comment:
What if we make GenericSearchProps also include the mantine Autocomplete props so you can override placeholder if you want to?